instrumentation/opentelemetry-instrumentation-threading/src/opentelemetry/instrumentation/threading/__init__.py

```diff
@@ -0,0 +1,160 @@
+            threading.Thread,
+            ThreadingInstrumentor.__WRAPPER_RUN_METHOD,
+            ThreadingInstrumentor.__wrap_threading_run,
+        )
+
+    @staticmethod
+    def _instrument_timer():
+        wrap_function_wrapper(
+            threading.Timer,
+            ThreadingInstrumentor.__WRAPPER_START_METHOD,
+            ThreadingInstrumentor.__wrap_threading_start,
+        )
+        wrap_function_wrapper(
+            threading.Timer,
+            ThreadingInstrumentor.__WRAPPER_RUN_METHOD,
+            ThreadingInstrumentor.__wrap_threading_run,
+        )
+
+    @staticmethod
+    def _instrument_thread_pool():
+        wrap_function_wrapper(
+            futures.ThreadPoolExecutor,
+            ThreadingInstrumentor.__WRAPPER_SUBMIT_METHOD,
+            ThreadingInstrumentor.__wrap_thread_pool_submit,
+        )
+
+    @staticmethod
+    def _uninstrument_thread():
+        unwrap(threading.Thread, ThreadingInstrumentor.__WRAPPER_START_METHOD)
+        unwrap(threading.Thread, ThreadingInstrumentor.__WRAPPER_RUN_METHOD)
+
+    @staticmethod
+    def _uninstrument_timer():
+        unwrap(threading.Timer, ThreadingInstrumentor.__WRAPPER_START_METHOD)
+        unwrap(threading.Timer, ThreadingInstrumentor.__WRAPPER_RUN_METHOD)
+
+    @staticmethod
+    def _uninstrument_thread_pool():
+        unwrap(
+            futures.ThreadPoolExecutor,
+            ThreadingInstrumentor.__WRAPPER_SUBMIT_METHOD,
+        )
+
+    @staticmethod
+    def __wrap_threading_start(call_wrapped, instance, args, kwargs):
+        instance._otel_context = context.get_current()
+        return call_wrapped(*args, **kwargs)
+
+    @staticmethod
+    def __wrap_threading_run(call_wrapped, instance, args, kwargs):
+        token = None
+        try:
+            token = context.attach(instance._otel_context)
+            return call_wrapped(*args, **kwargs)
+        finally:
+            context.detach(token)
+
+    @staticmethod
+    def __wrap_thread_pool_submit(call_wrapped, instance, args, kwargs):
+        # obtain the original function and wrapped kwargs
+        original_func = args[0]
+        wrapped_kwargs = {
+            ThreadingInstrumentor.__WRAPPER_KWARGS: kwargs,
+            ThreadingInstrumentor.__WRAPPER_CONTEXT: context.get_current(),
+        }
+
+        def wrapped_func(*func_args, **func_kwargs):
+            original_kwargs = func_kwargs.pop(
+                ThreadingInstrumentor.__WRAPPER_KWARGS
+            )
+            otel_context = func_kwargs.pop(
+                ThreadingInstrumentor.__WRAPPER_CONTEXT
+            )
+            token = None
+            try:
+                token = context.attach(otel_context)
+                return original_func(*func_args, **original_kwargs)
+            finally:
+                context.detach(token)
+
+        # replace the original function with the wrapped function
+        new_args = (wrapped_func,) + args[1:]
+        return call_wrapped(*new_args, **wrapped_kwargs)
```

Comment:
Would it work to just capture the OTel context in the closure of `wrapped_func` function? That way you don't need to modify the kwargs at all

```py
    def __wrap_thread_pool_submit(call_wrapped, instance, args, kwargs):
        # obtain the original function and calling context
        original_func = args[0]
        otel_context = context.get_current()

        def wrapped_func(*func_args, **func_kwargs):
            token = None
            try:
                token = context.attach(otel_context)
                return original_func(*func_args, **func_kwargs)
            finally:
                context.detach(token)

        # replace the original function with the wrapped function
        new_args = (wrapped_func,) + args[1:]
        return call_wrapped(*new_args, **kwargs)
```

Reply:
It sound great, thanks for your advice. I have updated it.